core/src/main/scala/cats/Apply.scala

```diff
@@ -224,35 +224,8 @@ trait Apply[F[_]] extends Functor[F] with InvariantSemigroupal[F] with ApplyArit
       val G = Apply[G]
     }
 
-  /**
-   * An `if-then-else` lifted into the `F` context.
-   * This function combines the effects of the `fcond` condition and of the two branches,
-   * in the order in which they are given.
-   *
-   * The value of the result is, depending on the value of the condition,
-   * the value of the first argument, or the value of the second argument.
-   *
-   * Example:
-   * {{{
-   * scala> import cats.implicits._
-   *
-   * scala> val b1: Option[Boolean] = Some(true)
-   * scala> val asInt1: Option[Int] = Apply[Option].ifA(b1)(Some(1), Some(0))
-   * scala> asInt1.get
-   * res0: Int = 1
-   *
-   * scala> val b2: Option[Boolean] = Some(false)
-   * scala> val asInt2: Option[Int] = Apply[Option].ifA(b2)(Some(1), Some(0))
-   * scala> asInt2.get
-   * res1: Int = 0
-   *
-   * scala> val b3: Option[Boolean] = Some(true)
-   * scala> val asInt3: Option[Int] = Apply[Option].ifA(b3)(Some(1), None)
-   * asInt2: Option[Int] = None
-   *
-   * }}}
-   */
   @noop
+  @deprecated("Dangerous method, use ifM (a flatMap) or ifF (a map) instead", "2.6.2")
   def ifA[A](fcond: F[Boolean])(ifTrue: F[A], ifFalse: F[A]): F[A] = {
```

Comment:
We can also eventually make this package private, shutting off the source access without breaking bincompat. We should wait a while though.